### PR TITLE
Compress barcode.fasta

### PR DESCRIPTION
**Pull request by FrickTobias**
Solves #154. Everything was compatible with reading .gz or from stdin.

### CODE REVIEW

File: src/blr/rules/trim.smk [hunk 1]
```diff
@@ -62,7 +62,7 @@ rule trim_and_tag:
 rule extract_DBS:
     # Extract barcode sequence from read1 FASTQ
     output:
-        fastq="barcodes.fasta",
+        fastq="barcodes.fasta.gz",
```
Comment (pontushojer) [inline src/blr/rules/trim.smk]:
```suggestion
        fastq="barcodes.fasta.gz"
```
Seems there is a rouge comma here.

File: src/blr/rules/trim.smk [hunk 2]
```diff
@@ -62,7 +62,7 @@ rule trim_and_tag:
...
     input:
         fastq="reads.1.fastq.gz",
```
Comment (pontushojer) [inline src/blr/rules/trim.smk]:
```suggestion
        fastq="reads.1.fastq.gz"
```
Also here

**Comment by pontushojer** [comment]
Short and sweet! 👍 

I will merge once the comments are handled.

**Comment by FrickTobias** [comment]
Hm, don't know why we had the commas there in the first place but I accepted your suggestions.

**Comment by marcelm** [comment]
It was probably me who added the trailing comma out of habit because I’ve started to do so in regular Python. See how [Black does it](https://test-black.readthedocs.io/en/style-guide/the_black_code_style.html#trailing-commas). In short, having trailing commas on every line in multi-line expressions reduces the diff when you add a new expression. But it’s totally fine to remove it here because we’re not being consistent anyway, so no worries.

**Pull request merged**